### ISSUE TITLE
Control not working anymore

### ISSUE THREAD

**Issue by TheDatadrainer** [pawlizio/my_velux#18]
Hello,
Recently I updated HASS IO and after the update restarted the Pi, the Velux covers were not working. From Home-Assistant Entities, each roller shutter (I have three) state is "Open" with nothing happening when using Up and Down commands.
I'm using Shutter Row cardboard add-on in the Overview panel and shows an icon state as open with a value of -24% for each roller shutter...
The connection to the KLF-200 is working. When restarting HASS IO, the reboot command is correctly sent as the LED on the KLF is blinking. And there is not error on the log "custom_components.velux" set to debug in configuration.yaml.
Can a change in Home-Assistant have produce this? Thanks.

**Comment by pawlizio** [comment]
Please enable logging and provide an extract of the log files. May they help to identify the issue: 

Enable logger in your configuration.yaml file for pyvlx and custom_component.velux as shown below: 
```
logger:
  default: warning
  logs:
    pyvlx: debug
    custom_components.velux: debug
```

BR

**Comment by TheDatadrainer** [comment]
Thank you for your answer.
Here is my version of Home-Assistant:
    Home Assistant: 2022.11.4
    Supervisor: 2022.10.2
    Operating System: 9.3
    User interface: 20221108.0 - latest
and attached is the log of today after the last restart.
[my_velux.log](https://github.com/pawlizio/my_velux/files/10070166/my_velux.log)

As I said I didn't see anything wrong in the log. But I hope will be able to help me to find what goes wrong.
I removed the cover entities, cleaned everything and removed the plugin. Reinstalled it and configured with no problem. But still no control working... But with the state now being always Closed instead of Open and sending 0% instead of -24% from the cardboard. When I try to use the slider, it stay where I put it, but nothing is happening .
![control_failing](https://user-images.githubusercontent.com/56295922/203397193-75721f7a-1ec6-4570-89eb-a58b198678b7.jpg)

**Comment by dumpfheimer** [comment]
I believe KLF opens a WiFi hotspot when freshly booted, and that there might be a possibility to controll the blinds from there. You could try that to rule out the possibility that something is wrong between the gateway and your blind.
You could also try rolling back to the last known working version and check if everything goes back to normal

**Comment by TheDatadrainer** [comment]
Maybe, indeed the KLF-200 have problem. I need to check that. For me the problem was not from the KLF-200 it is able to list correctly the registered roller shutters. But who knows. I'll try to reconfigure or if not possible to reset the KLF-200 and will let you know.
Rolling back the version of HASS.io is not an option.
I you have any idea, it's welcome.
Did the log had something strange?

**Comment by pawlizio** [comment]
So from the log everything looks fine so far. Actually the KLF-200 also sends notifications abouts the command requests and session finished notifications for each command. Strange that the actuator does not move. Do you have other remotes to control the shutters, are they working? It does not look like a problem between HA and KLF200.

**Comment by dumpfheimer** [comment]
No the devices/blindes are stored on the device and would show up even if there are connection issues between the gateway and the device itself.

I would start with the "WiFi hotspot control Test". You could also try moving the gateway closer to the blinds, maybe something is disturbing the signal?

**Comment by TheDatadrainer** [comment]
> Do you have other remotes to control the shutters, are they working? It does not look like a problem between HA and KLF200.

Yes. I have a KLR-200 (touch screen) and three KLI-310 (one square wall remote and two as traditional remote control).
Do you think the problem can came from one? They are all working fine. If I remember well the KLF-200 detected and registered everything even the KLR-200 and a Somfy smart plug that I can control from pyvelux but not from Home-Assistant. That was a few years ago and my memory is not perfect.
The KLF-200 is at ~5 m of two blinds and ~8 m from the last one with nothing except wood and bricks in-between.
As I said, I will try to connect to its web GUI to see if something bad happened. And I will try a reset and registering everything back. I'll tell you the result. But if everything is fine with it. I will have to check.
But do you confirm my_velux add-on is working fine with the latest version of Home-Assistant? Thanks!

**Comment by TheDatadrainer** [comment]
I checked the KLF-200 using its web service and found nothing odd. I was ready to reset it, but before that I wanted to test it again. And to my surprise here is what happened [home-assistant.zip](https://github.com/pawlizio/my_velux/files/10078660/home-assistant.zip). The roller shutters responded with a delay of ~20s after sending a command, but responded correctly...
Tested the thing during several minutes, alternating remote control commands with Home-Assistant in the hope to have a clue and got nothing except immediate response from the remotes and the delay with HA with the some command being not received in the right order, I think because of the delay in HA.
Then I rebooted to see if that's changed anything and it does... It came back to the same behavior as before and that bring me here.
So I think it is definitely not a KLF-200 issue. I remember what @pawlizio said about the other devices (cg. remotes control and a smart plug for me). I check the batteries, especially the KLR-200 that can have a odd behavior when the batteries begin to die. But they are all good and the hardware is perfectly working. No problem here too. No signal issue too, as I moved the KLF-200 near two of the KUX-110 to be sure (One of the roller shutter is solar, the two other are wired, controller by KUX-110 controllers). So can be a problem with HA, but where?
Maybe there is something to learn from this log. I spend sometime on it, but does not find anything to help me understand what was happening.
Thank you very much for your time. That's very nice.

**Comment by pawlizio** [comment]
@TheDatadrainer , many thanks for that logs. I found a not-awaited heartbeat stop command, in case the heartbeat restarted. That's what you can see in your logs, a heartbeat restarts, but just a new starts while the old is still running in parallel. I changed that on pyvlx side, a restart of HA shall load the modified pyvlx version 0.1.31 which has some heartbeat modification.  Should look like this during HA start in the log:
```
2022-11-23 22:26:10.135 DEBUG (MainThread) [pyvlx] Heartbeat __init__
2022-11-23 22:26:10.135 DEBUG (MainThread) [pyvlx] Loading https://github.com/pawlizio/pyvlx.git@master v0.1.31
```
However I don't think that this solves your problem with your non responding shutter. Heartbeat just ensures that the communication to KLF stays open.

**Comment by TheDatadrainer** [comment]
Hi there.
Solved my problem sometime now. It has nothing to do with the KLF-200 or Home-Assistant but with a computer under GNU/Linux where the KLF-200 was plugged to... I had completely forgotten that but as the KLF need less than 500 mA, to work and this computer have many USB ports with charging capabilities that I didn't used, so I put it there. But it seems an OS update have changed things, as there was no problem before to update it. I reverted to the original USB power adapter instead. And everything is fine since. So I'm closing this issue.
Thank you for your help.
